packages/mui-joy/src/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,281 @@
+      }),
+      ...(ownerState.size === 'md' && {
+        '--Avatar-size': '4rem',
+      }),
+      ...(ownerState.size === 'lg' && {
+        '--Avatar-size': '5rem',
+      }),
+      position: 'relative',
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
+      flexShrink: 0,
+      width: 'var(--Avatar-size)',
+      height: 'var(--Avatar-size)',
+      lineHeight: 1,
+      borderRadius: '50%',
+      overflow: 'hidden',
+      userSelect: 'none',
+      ...(ownerState.shape === 'rounded' && {
+        borderRadius: theme.vars.radius[ownerState.size!],
+      }),
+      ...(ownerState.shape === 'square' && {
+        borderRadius: 0,
+      }),
+    },
+    theme.variants[ownerState.variant!]?.[ownerState.color!],
+    theme.variants[`${ownerState.variant!}Hover`]?.[ownerState.color!],
+    theme.variants[`${ownerState.variant!}Active`]?.[ownerState.color!],
+    theme.variants[`${ownerState.variant!}Disabled`]?.[ownerState.color!],
+  ];
+});
+
+const AvatarImg = styled('img', {
+  name: 'MuiAvatar',
+  slot: 'Img',
+  overridesResolver: (props, styles) => styles.img,
+})(() => {
+  return [
+    {
+      width: '100%',
+      height: '100%',
+      textAlign: 'center',
+      // Handle non-square image. The property isn't supported by IE11.
+      objectFit: 'cover',
+      // Hide alt text.
+      color: 'transparent',
+      // Hide the image broken icon, only works on Chrome.
+      textIndent: 10000,
+    },
+  ];
+});
+
+const AvatarFallback = styled(Person, {
+  name: 'MuiAvatar',
+  slot: 'Fallback',
+  overridesResolver: (props, styles) => styles.fallback,
+})({
+  width: '75%',
+  height: '75%',
+});
+
+type UseLoadedProps = { src?: string; srcSet?: string; crossOrigin?: any; referrerPolicy?: any };
+
+function useLoaded({ crossOrigin, referrerPolicy, src, srcSet }: UseLoadedProps) {
+  const [loaded, setLoaded] = React.useState<string | boolean>(false);
+
+  React.useEffect(() => {
+    if (!src && !srcSet) {
+      return undefined;
+    }
+
+    setLoaded(false);
+
+    let active = true;
+    const image = new Image();
+    image.onload = () => {
+      if (!active) {
+        return;
+      }
+      setLoaded('loaded');
+    };
+    image.onerror = () => {
+      if (!active) {
+        return;
+      }
+      setLoaded('error');
+    };
+    image.crossOrigin = crossOrigin;
+    image.referrerPolicy = referrerPolicy;
+    if (src) {
+      image.src = src;
+    }
+    if (srcSet) {
+      image.srcset = srcSet;
+    }
+
+    return () => {
+      active = false;
+    };
+  }, [crossOrigin, referrerPolicy, src, srcSet]);
+
+  return loaded;
+}
+
+const Avatar = React.forwardRef(function Avatar(inProps, ref) {
+  const props = useThemeProps<typeof inProps & AvatarProps>({
+    props: inProps,
+    name: 'MuiAvatar',
+  });
+
+  const {
+    alt,
+    className,
+    color = 'neutral',
+    component = 'div',
+    size = 'md',
+    shape = 'circular' as AvatarShapeProp,
```

Comment:
Otherwise, I get the following error when passing `ownerState` to `AvatarRoot`. Do you know a better way to handle this?

```js
Types of property 'shape' are incompatible.
Type 'string' is not assignable to type 'AvatarShapeProp | undefined'
```

Reply:
It's weird that I get this type error only for `shape` while `size` and `variant` are fine.